### ISSUE TITLE
`deno run` cannot find export in module that has export

### ISSUE THREAD

**Issue by dsherret** [denoland/deno#12755]
The latest 1.16.1 has issues where it says that a module doesn't have an export when it clearly does. It is very difficult to reproduce this and so I haven't been able to create a simple example yet, but it's happened to me twice today during my development so I'm logging it (actually 3 times, but the third was slightly different and probably related).

The script is using this import:

```ts
import { createMinifier } from "https://deno.land/x/dts_minify@0.3.0/mod.ts";
```

Then see what occurred on my command line... the issue is fixed when I run it in an old version of deno (in this case 1.15):

```ts
> deno run -A scripts/createLibFile.ts
warning: Cannot load module "https://deno.land/x/dts_minify@0.3.0/mod.ts".
    at file:///V:/ts-morph/packages/common/scripts/createLibFile.ts:1:32
  If the source module contains only types, use `import type` and `export type` to import it instead.
error: Uncaught SyntaxError: The requested module 'https://deno.land/x/dts_minify@0.3.0/mod.ts' does not provide an export named 'createMinifier'
import { createMinifier } from "https://deno.land/x/dts_minify@0.3.0/mod.ts";
         ~~~~~~~~~~~~~~
    at <anonymous> (file:///V:/ts-morph/packages/common/scripts/createLibFile.ts:1:10)
> deno -V
deno 1.16.1
> bvm exec deno 1.15 run -A scripts/createLibFile.ts
Check file:///V:/ts-morph/packages/common/scripts/createLibFile.ts
> deno run -A scripts/createLibFile.ts
Check file:///V:/ts-morph/packages/common/scripts/createLibFile.ts
>
```

The issue can also be fixed by deleting the deno cache folder.

This might be related to #12740.

**Comment by andreubotella** [comment]
A similar issue, also caused by the cache being messed up, was reported by a few users on Discord. Here's the output posted by @sno2:

```
$ deno run -A --unstable --reload dev.ts
Download https://deno.land/x/typedffi@v0.1.2/mod.ts
Check file:///C:/projects/deno/bertml/dev.ts
error: Emitted code for "file:///C:/projects/deno/bertml/dev.ts" not found.
```

**Comment by KyleJune** [comment]
I've encountered this issue with `deno test` too. I had the --watch flag on and moved one of my external dependencies from one file to another (deps.ts). deps.ts exported the external module and I updated the test code to import from it. I resolved the issue by re-running with the --reload flag. I don't have a clear reproduction path, just wanted to share that it's not just `deno run` in case that may help with tracking down the bug.
